src/foam/nanos/auth/AuthenticatedUserDAO.java

```diff
@@ -60,4 +56,34 @@ public FObject put_(X x, FObject fObject) {
       return null;
     }
   }
+
+  @Override
+  /**
+   * Restrict find method to only return the logged in user object
+   * */
+  public FObject find_(X x, Object id) {
+    User user = (User) x.get("user");
+
+    if ( user == null ) {
```

Comment:
I think this should check if user == null or id != user.getId(). Otherwise someone could try to lookup someone else and think they've succeeded.